opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -0,0 +1,538 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2014-2018 Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2018      Intel, Inc, All rights reserved
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+
+#include "opal_config.h"
+
+#include "opal/mca/btl/btl.h"
+#include "opal/mca/btl/base/base.h"
+#include "opal/mca/hwloc/base/base.h"
+
+#include <string.h>
+
+#include "btl_ofi.h"
+#include "btl_ofi_providers.h"
+#include "btl_ofi_rdma.h"
+
+
+static char *prov_include;
+static char *prov_exclude;
+static char *ofi_progress_mode;
+static int mca_btl_ofi_init_device(struct fi_info *info);
+
+static int mca_btl_ofi_component_register(void)
+{
+    mca_btl_ofi_module_t *module = &mca_btl_ofi_module_template;
+
+    /* fi_getinfo with prov_name == NULL means ALL provider.
+     * Since now we are using the first valid info returned, I'm not sure
+     * if we need to provide the support for comma limited provider list. */
+    prov_include = NULL;
+    (void) mca_base_component_var_register(&mca_btl_ofi_component.super.btl_version,
+                                          "provider_include",
+                                          "OFI provider that are considered for use "
+                                          "(e.g., \"psm,psm2\"; an empty value means that all providers will "
+                                          "be considered.",
+                                          MCA_BASE_VAR_TYPE_STRING, NULL, 0, 0,
+                                          OPAL_INFO_LVL_1,
```

Comment:
These should not be level 1.  See https://github.com/open-mpi/ompi/wiki/MCAParamLevels.

Reply:
Ok.  Note that I didn't specifically call out each of the MCA params, but you might want to look at the levels of all your MCA params in the context of that wiki page.  The idea is that we want very, very few things in levels 1,2,3.

Reply:
I fixed all of the param and they are at least level 4.